New translation - Hungarian

AttilaSVK opened the pull request:
- Hungarian / Hungary
 - Maintainers head: @AttilaSVK
- Co-maintainers:
- [x] Active      since January 2022
- [X] Maintained  since January 2022

PFW-1283

AttilaSVK committed:
New translation

Added Hungarian language

AttilaSVK committed:
Update hu.po

Corrected two ...(one character) to ...(three characters) and removed an accidental accent to maintain ASCII compatibility

3d-gussner commented on lang/po/new/hu.po:
```diff
@@ -2,1771 +2,1767 @@
 #: ultralcd.cpp:4838
 msgid "Now I will calibrate distance between tip of the nozzle and heatbed surface."
-msgstr ""
+msgstr "Most beallítjuk a fuvoka hegye es a targyasztal felulete kozotti tavolsagot."
```
The `lang-import.sh hu` fails as `beallítjuk` contains a non-ascii character.
Please change this.

3d-gussner reviewed (commented):
Thanks for the translation.

Is there any default rule how to replace characters with diacritics to ascii only?
See [lang-import.sh](https://github.com/AttilaSVK/Prusa-Firmware/blob/MK3_3.11.0/lang/lang-import.sh#L201-L208) If you know how to change it please add it to that file, otherwise comment here the rules and we can add these.

3d-gussner commented:
Also you have exceeded some limits defined by `c=xx r=yy` 
Here the results of `./lang-check.py hu` after replacing the `í` with an `i`
```
/Prusa-Firmware/MK3_3.11.0-Hungarian/lang$ ./lang-check.py hu
Start hu lang-check
[E]: Text is longer than definition on line 7: cols=18 rows=1 (rows diff=1)
 source text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇
   |FS v0.3 or older  |
 translated text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇
   |FS v0.3 vagy regeb|bi

[E]: Text is longer than definition on line 35: cols=15 rows=1 (rows diff=1)
 source text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄
   |Adjusting Z:   |
 translated text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄
   |Z magassag alli|tasa:

[E]: Text is longer than definition on line 51: cols=6 rows=1 (rows diff=1)
 source text:
    ₀₁₂₃₄₅
   |Auto  |
 translated text:
    ₀₁₂₃₄₅
   |Automa|ta

[E]: Text is longer than definition on line 59: cols=20 rows=2 (rows diff=1)
 source text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Are left and right  |
   2 |Z~carriages all up? |
 translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |A Z tengely mindket |
   2 |oldala a felso      |
   3 |vegponton van?      |

[E]: Text is longer than definition on line 71: cols=20 rows=4 (rows diff=1)
 source text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Autoloading filament|
   2 |available only when |
   3 |filament sensor is  |
   4 |turned on...        |
 translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Filament automata   |
   2 |betolteste csak     |
   3 |bekapcsolt filament |
   4 |szenzorral erheto   |
   5 |el...               |

[E]: Text is longer than definition on line 75: cols=20 rows=4 (rows diff=1)
 source text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Autoloading filament|
   2 |is active, just     |
   3 |press the knob and  |
   4 |insert filament...  |
 translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Automata betoltes   |
   2 |bekapcsolva, csak   |
   3 |nyomd meg a gombot  |
   4 |es helyzed be a     |
   5 |filamentet...       |

[E]: Text is longer than definition on line 127: cols=20 rows=2 (rows diff=1)
 source text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Blackout occurred.  |
   2 |Recover print?      |
 translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Aramkieses volt,    |
   2 |nyomtatas           |
   3 |folytatasa?         |

[E]: Text is longer than definition on line 251: cols=20 rows=1 (rows diff=1)
 source text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   |Endstop not hit     |
 translated text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   |Vegallask. nem kapcs|.

[E]: Text is longer than definition on line 279: cols=20 rows=4 (rows diff=1)
 source text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |ERROR: Filament     |
   2 |sensor is not       |
   3 |responding, please  |
   4 |check connection.   |
 translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |HIBA: A filament    |
   2 |szenzor nem         |
   3 |valaszol, kerlek    |
   4 |ellenorizd a        |
   5 |csatlakozast.       |

[E]: Text is longer than definition on line 335: cols=20 rows=2 (rows diff=1)
 source text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Filament extruding &|
   2 |with correct color? |
 translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Extrudalodik a      |
   2 |filament? Helyes    |
   3 |szinnel?            |

[E]: Text is longer than definition on line 387: cols=14 rows=1 (rows diff=1)
 source text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃
   |Front side[um]|
 translated text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃
   |Elulso old.[um|]

[E]: Text is longer than definition on line 439: cols=20 rows=1 (rows diff=1)
 source text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   |Checking sensors    |
 translated text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   |Szenzorok ellenorzes|e

[E]: Text is longer than definition on line 479: cols=20 rows=1 (rows diff=1)
 source text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   |Insert filament     |
 translated text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   |Helyezd be a filamen|tet

[E]: Text is longer than definition on line 587: cols=14 rows=1 (rows diff=1)
 source text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃
   |Measured skew |
 translated text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃
   |Mert merol. hi|ba

[E]: Text is longer than definition on line 795: cols=20 rows=3 (rows diff=2)
 source text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Please remove       |
   2 |shipping helpers    |
   3 |first.              |
 translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Kerlek, eloszor     |
   2 |tavolitsd el a      |
   3 |szallitashoz        |
   4 |szukseges           |
   5 |segedanyagokat.     |

[E]: Text is longer than definition on line 887: cols=20 rows=4 (rows diff=1)
 source text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Please check the IR |
   2 |sensor connection,  |
   3 |unload filament if  |
   4 |present.            |
 translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Kerlek nezd meg az  |
   2 |IR erzekelo         |
   3 |csatlakozasat, vedd |
   4 |ki a filamentet, ha |
   5 |bent van.           |

[E]: Text is longer than definition on line 891: cols=20 rows=1 (rows diff=1)
 source text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   |Recovering print    |
 translated text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   |Nyomtatas visszaalli|tasa.

[E]: Text is longer than definition on line 1075: cols=20 rows=4 (rows diff=1)
 source text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Select temperature  |
   2 |which matches your  |
   3 |material.           |
 translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Valassz             |
   2 |homersekletet, ami  |
   3 |megfelel a          |
   4 |filamenthez ajanlott|
   5 |homersekletnek.     |

[E]: Text is longer than definition on line 1091: cols=20 rows=3 (rows diff=1)
 source text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Sensor verified,    |
   2 |remove the filament |
   3 |now.                |
 translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |Szenzor             |
   2 |leellenorizve, vedd |
   3 |ki a filamentet     |
   4 |most.               |

[E]: Text is longer than definition on line 1123: cols=20 rows=1 (rows diff=1)
 source text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   |to load filament    |
 translated text:
    ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   |filament betoltesehe|z

[E]: Text is longer than definition on line 1199: cols=3 rows=1 (rows diff=1)
 source text:
    ₀₁₂
   |Yes|
 translated text:
    ₀₁₂
   |Ige|n

[W]: Differing last punctuation character (! => .) on line 1211:
 source text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |XYZ calibration all |
   2 |right. X/Y axes are |
   3 |slightly skewed.    |
   4 |Good job!           |
 translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |XYZ kalibracio      |
   2 |sikerult. Az X/Y    |
   3 |tengelyeken enyhe   |
   4 |merolegessegi hiba  |
   5 |van.                |

[W]: Differing last punctuation character (! => .) on line 1223:
 source text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |XYZ calibration ok. |
   2 |X/Y axes are        |
   3 |perpendicular.      |
   4 |Congratulations!    |
 translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |XYZ kalibracio      |
   2 |sikerult. Az X/Y    |
   3 |tengelyek           |
   4 |merolegesek.        |

End hu lang-check
```

3d-gussner commented:
You don't have to change the `Yes` / `Igen` as in this case we should be able expand the limit to `c=4`

AttilaSVK committed:
Update hu.po

Reduced long strings

AttilaSVK committed:
Update hu.po

Revised some translations based on the output from lang-check.py

AttilaSVK commented:
Corrected all the translations based on the results of lang-check.py (finally got lang-import.sh to work, that's why I couldn't run lang-check.py). The only error is Yes/Igen.

AttilaSVK committed:
Update hu.po

Corrected a string based on where it appears on the display (Main, corrected to Főmenü, meaning Main Menu)

AttilaSVK commented:
Corrected one more string based on the firmware I compiled and flashed on my printer, now this should be good enough for release. Regarding character replacement for ASCII compatibility, it's the same as in Czech with the vowels, just strip the accents (umlauts and double accents in Hungarian), so á, é, í, ó, ö, ő, ú, ü, ű becomes a, e, i, o, o, o, u, u, u.
There are no non-ASCII consonants in Hungarian.

3d-gussner commented:
Thanks for the update and here the result of `./lang-check.py hu`
```
/Prusa-Firmware/MK3_3.11.0-Romanian/lang$ ./lang-check.py hu
Start hu lang-check
[E]: Text is longer than definition on line 1199: cols=3 rows=1 (rows diff=1)
 source text:
    ₀₁₂
   |Yes|
 translated text:
    ₀₁₂
   |Ige|n

[W]: Differing last punctuation character (! => .) on line 1211:
 source text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |XYZ calibration all |
   2 |right. X/Y axes are |
   3 |slightly skewed.    |
   4 |Good job!           |
 translated text:
      ₀₁₂₃₄₅₆₇₈₉₀₁₂₃₄₅₆₇₈₉
   1 |XYZ kalibracio      |
   2 |sikerult. Az X/Y    |
   3 |tengelyeken enyhe   |
   4 |merolegessegi hiba  |
   5 |van.                |

End hu lang-check
```

As said, we will expand the `Yes/No` message limits to `c=4` 
Is there a reason why you don't use the `!` in the 2nd [W]arning?

3d-gussner commented:
Do you want a firmware hex file to test it?
Here a preview in the MK404 Prusa MK3S simulator:
![image](https://user-images.githubusercontent.com/25530011/149278027-e5d668e6-e3de-4795-a68c-dceae562f94f.png)

AttilaSVK commented:
Thanks, I've managed to get the PF-build.sh script working, and it did produce me a firmware with Hungarian language working, which I've tested on my MK3S.
Regarding the different punctuation character, I didn't translate "Good job!" to keep the text short, and it's not something you'd say in Hungarian. However I can add the Hungarian equivalent "Nice job!", which is "Szep munka!". I think that could still fit in 5 lines. If you want to have it added, just tell me and I'll add it to the translation file.

3d-gussner commented:
Great that you could compile the firmware yourself. The messages with more than r=5 can be longer as they are shown on multiple screens. We can play a tiny bit with the rows here.
That doesn't mean you can write a novel here😉.
Depending on the text it can even look better to have 6-7 rows instead of 5.

As the firmware team is always looking for more free resources we should try to stay within the limits and even improve to save space.

3d-gussner commented:
@AttilaSVK We have found an issue with the limits of one message and it has been merged with https://github.com/prusa3d/Prusa-Firmware/pull/3348. Can you please `git rebase` your branch / PR.

Sorry for the inconvenience.

Do you know anyone that would be willing to review your translation and approve it?

The goal is to have at least two maintainers checking each others translations.

AttilaSVK commented:
Regarding the rebase, I think I did it right, I'm new to git tools.
I'll ask around for a co-maintainer.

3d-gussner commented:
The rebase is looking good. Well done.

As you were able to compile the firmware and used `./lang-impor.sh hu` I would like to suggest that you also provide two more files `lang/po/Firmware_hu.po` and `/lang/lang_en_hu.txt` as well.

- Change to `lang` folder
- run `./lang-import.sh hu` which will generate the `lang/po/new/lang_en_hu.txt` file
- replace existing `lang/lang_en_hu.txt` with `lang/po/new/lang_en_hu.txt`
- run `./lang-check.py hu` to verify that your new `lang_en_hu.txt` is okay.
- delete `lang/po/new/nonasci.txt , hu_filtered.po , hu_new.po`
- run `./lang-export.sh hu` which will update the `lang/Firmware_hu.po` file.

- commit your changes and update your branch and PR with a push

AttilaSVK committed:
Updated Hungarian translation files

updated lang/lang_en_hu.txt and lang/po/Firmware_hu.po with files generated by lang-import.sh and lang-export.sh

AttilaSVK commented:
Should be done

3d-gussner commented:
@AttilaSVK can you have a look at the replacement I want to add to `lang-import.sh`
```
if [ "$LNG" = "hu" ]; then`# See https://www.fileformat.info/info/charset/UTF-8/list.htm
 #replace 'Á' with 'A'(acute)
 sed -i 's/\xc3\x81/A/g' $LNG'_filtered.po'
 #replace 'á' with 'a'
 sed -i 's/\xc3\xa1/a/g' $LNG'_filtered.po'
 #replace 'É' with 'E' (acute)
 sed -i 's/\xc3\x89/E/g' $LNG'_filtered.po'
 #replace 'é' with 'e'
 sed -i 's/\xc3\xa9/e/g' $LNG'_filtered.po'
 #replace 'Í' with 'I' (acute)
 sed -i 's/\xc3\x8d/I/g' $LNG'_filtered.po'
 #replace 'i̇́' with 'i'
 sed -i 's/\xc3\xad/i/g' $LNG'_filtered.po'
 #replace 'Ó' with 'O' (acute)
 sed -i 's/\xc3\x93/O/g' $LNG'_filtered.po'
 #replace 'ó' with 'o'
 sed -i 's/\xc3\xb3/o/g' $LNG'_filtered.po'
 #replace 'Ö' with 'O' (diaresis)
 sed -i 's/\xc3\x96/O/g' $LNG'_filtered.po'
 #replace 'ö' with 'o'
 sed -i 's/\xc3\xb6/o/g' $LNG'_filtered.po'
 #replace 'Ő' with 'O' (double acute)
 sed -i 's/\xc5\x90/O/g' $LNG'_filtered.po'
 #replace 'ő' with 'o'
 sed -i 's/\xc5\x91/o/g' $LNG'_filtered.po'
 #replace 'Ú' with 'U' (acute) 
 sed -i 's/\xc3\x9a/U/g' $LNG'_filtered.po'
 #replace 'ú' with 'u'
 sed -i 's/\xc3\xba/u/g' $LNG'_filtered.po'
 #replace 'Ü' with 'U' (diaersis)
 sed -i 's/\xc3\x9c/U/g' $LNG'_filtered.po'
 #replace 'ü' with 'u'
 sed -i 's/\xc3\xbc/u/g' $LNG'_filtered.po'
 #replace 'Ű' with 'U' (double acute)
 sed -i 's/\xc5\xb0/U/g' $LNG'_filtered.po'
 #replace 'ű' with 'u'
 sed -i 's/\xc5\xb1/u/g' $LNG'_filtered.po'
fi
``

3d-gussner commented:
@AttilaSVK See also https://github.com/prusa3d/Prusa-Firmware/pull/3342/files#diff-a24289c9ad15eb88af1103e3112ffcd9dca5beb53390bd094ddac06fa1e792f3R238-R274

Hauzman commented:
@AttilaSVK I can have a lock over it today if you did not find a second person, I have a lock over it and it's locking good but did not have time to test it on the printer

AttilaSVK commented:
@3d-gussner I don't know the syntax to sed (should learn it though), but according to the comments it looks good.
@Hauzman it would be great if you had a look at the text file. The firmware compiles fine with it and runs fine both in MK404 and a real MK3S printer.

Hauzman commented on lang/lang_en_hu.txt:
```diff
@@ -1,1403 +1,1403 @@
 #MSG_MEASURED_SKEW c=14
 "Measured skew"
-"\x00"
+"Meroleg. hiba"
```
@AttilaSVK it nem lenne inkab job "Mert ferdeseg" vagy "Mert torzulas" ez csak javaslat :)

AttilaSVK replied:
A merőlegességi hiba pontosabb megfogalmazás szerintem.

Hauzman replied:
> A merőlegességi hiba pontosabb megfogalmazás szerintem.
@AttilaSVK igen azert montam hogy csak javaslat

3d-gussner replied:
After you agree on a translations, please comment it in English so we can read it as well.

AttilaSVK replied:
@3d-gussner will do in the future.
Regarding this translation, instead of skew I used the term perpendicularity error. (it sounds better in Hungarian IMHO, and it’s easier to shorten nicely)

Hauzman commented on lang/lang_en_hu.txt:
```diff
@@ -1,1403 +1,1403 @@
 #MSG_SEVERE_SKEW c=14
 "Severe skew"
-"\x00"
+"NagyMerol.hiba"
```
@AttilaSVK it ugyan az mint "Súlyos ferdeség"

AttilaSVK replied:
Ugyanaz a véleményem, mint fentebb.

Hauzman commented:
@AttilaSVK @3d-gussner Hi I have checked and I have 2 suggestion only but it's not necessary need to be changed this is my point of view, and don't know if you slowed/ changed  the issue where there is the #MSG_YES c=3/ "Yes" line 1197 in Hungarian is with 4  , in rest looking good.
I will try today to upload also on the printer and check it live to .

3d-gussner commented:
> @AttilaSVK @3d-gussner Hi I have checked and I have 2 suggestion only but it's not necessary need to be changed this is my point of view, and don't know if you slowed/ changed the issue where there is the #MSG_YES c=3/ "Yes" line 1197 in Hungarian is with 4 , in rest looking good. I will try today to upload also on the printer and check it live to .

We gonna change `Yes/No` to `c=4` as we have the space for it, but to prevent github merge conflicts we will create a new PR just for that after the translations PRs have been merged.

Hauzman commented:
From my perspective is looking good and understandable the translation

3d-gussner commented:
Approved by @Hauzman and the [build check](https://app.travis-ci.com/github/prusa3d/Prusa-Firmware/builds/244977252#L2765-L2789) is looking good as well.

The lang-check Error message will be solved with https://github.com/prusa3d/Prusa-Firmware/pull/3342

Thanks @AttilaSVK for the translation and PR and to @Hauzman for reviewing it and approval.
